### PR TITLE
Add dedicated plugin type struct

### PR DESCRIPTION
**Pull request by saschagrunert**
Follow-up on: https://github.com/containerd/nri/pull/121#discussion_r1872009094

PTAL @mikebrow

### CODE REVIEW

File: pkg/adaptation/plugin_type.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+import (
+	"context"
+
+	"github.com/containerd/nri/pkg/api"
+)
+
+type pluginType struct {
+	wasmImpl api.Plugin
+	grpcImpl api.PluginService
```
Comment (klihub) [inline pkg/adaptation/plugin_type.go:27]: nit: technically `ttrpcImpl` would be more correct.

File: pkg/adaptation/plugin_type.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+func (p *pluginType) isWasm() bool {
+	return p.wasmImpl != nil
+}
+
+func (p *pluginType) isGrpc() bool {
```
Comment (klihub) [inline pkg/adaptation/plugin_type.go:34]: nit: ditto here, strictly speaking `isTtrpc()` would be the correct name.

File: pkg/adaptation/plugin_type.go (hunk 1 above)
Comment (saschagrunert) [reply pkg/adaptation/plugin_type.go]: Makes sense, fixed as suggested :+1:

**Review by klihub — APPROVED** [review]
LGTM

**Review by mikebrow — COMMENTED** [review]
excellent

**Pull request merged**